services/src/handlers/workflows.rs

```diff
@@ -49,6 +52,37 @@ async fn load_workflow<C: Context>(id: Uuid, ctx: C) -> Result<impl warp::Reply,
     Ok(warp::reply::json(&wf).into_response())
 }
 
+pub(crate) fn get_workflow_metadata_handler<C: Context>(
+    ctx: C,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    warp::get()
+        .and(warp::path!("workflow" / "metadata" / Uuid))
+        .and(authenticate(ctx))
+        .and_then(get_workflow_metadata)
+}
+
+// TODO: move into handler once async closures are available?
+async fn get_workflow_metadata<C: Context>(
+    id: Uuid,
+    ctx: C,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    // TODO: check for rights to access workflow?
+    let workflow = ctx
+        .workflow_registry_ref()
+        .await
+        .load(&WorkflowId(id))
+        .await?;
+    let operator = workflow.operator.get_vector().context(error::Operator)?;
+    // TODO: use meaningful context to initialize
+    let operator = operator
+        .initialize(&MockExecutionContext::default())
+        .context(error::Operator)?;
+
+    let result_descriptor = operator.result_descriptor();
```

Comment:
we should think about caching this. maybe we can calculate it once on workflow creation? It doesn't depend on the query rectangle.

Reply:
I added a todo

7075bd4 

caching it would mean we compiled it ;-)